tests/integration/targets/rabbitmq_vhost/tasks/tests_api.yml

```diff
@@ -0,0 +1,269 @@
+- block:
+  - set_fact:
+      vhost_name: /test  
+
+  - name: Add host (check_mode)
+    rabbitmq_vhost:
+      name: "{{ vhost_name }}"
+      state: present
+      login_user: guest 
+      login_password: guest 
+      login_host: 127.0.0.1
+    register: result_check_mode
+    check_mode: yes
+
+  - name: Assert check_mode result for adding host
+    assert:
+      that:
+        - result_check_mode is success
+
+  - name: Add host
+    rabbitmq_vhost:
+      name: "{{ vhost_name }}"
+      state: present
+      login_user: guest 
+      login_password: guest 
+      login_host: 127.0.0.1
+    register: result
+
+  - name: Check that the host was created successfully
+    shell: "rabbitmqctl list_vhosts name tracing | grep {{ vhost_name }}"
```

Comment:
instead of `shell`, could we test this using the Ansible module `state=present` and just add a var to set `login_host` to `None` to force this module to fall back to `rabbitmqctl`?